core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -459,8 +459,21 @@ default void renameColumn(ConnectorSession session, ConnectorTableHandle tableHa
 
     /**
      * Drop the specified column
+     *
+     * @deprecated use {@link #dropColumn(ConnectorSession, ConnectorTableHandle, ColumnHandle, List)}
      */
+    @Deprecated
     default void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column)
+    {
+        dropColumn(session, tableHandle, column, List.of());
+    }
+
+    /**
+     * Drop the specified column
+     *
+     * @param field path to a field without the parent column name in a row type. DROP COLUMN a.b.c returns ['b', 'c'].
+     */
+    default void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column, List<String> field)
```

Comment:
it's counter-intuitive that empty `[]` field list means "drop whole column"
let's have this as a separate `dropField`  and promise the connector the list is never empty.

Also, we should document how the navigation works in case of arrays and maps.
eg. if my column has `array(row(a int, b int))` type, how do i drop the `b` field?